test/jdk/java/awt/Scrollbar/AquaLFScrollbarTest/ScrollBarBorderTest.java

```diff
@@ -0,0 +1,190 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Insets;
+import java.awt.image.BufferedImage;
+import java.awt.image.ImageObserver;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import javax.imageio.ImageIO;
+import javax.swing.JScrollBar;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+import javax.swing.border.Border;
+
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB;
+
+/*
+ * @test
+ * @bug 8190264
+ * @summary JScrollBar ignores its border when using macOS Mac OS X Aqua look and feel
+ * @run main ScrollBarBorderTest
+ */
+public class ScrollBarBorderTest {
+    private static JScrollBar scrollBar;
+    public static final int BORDER_WIDTH = 150;
+    public static final int WIDTH = BORDER_WIDTH + 200;
+    public static final int HEIGHT = 20;
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+            System.out.println(laf.getName());
+        } catch (UnsupportedLookAndFeelException ignored){
+            System.out.println("Unsupported LookAndFeel: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException |
+                IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void testHorizontal() {
+        // create scroll bar
+        scrollBar = new JScrollBar(JScrollBar.HORIZONTAL);
+        scrollBar.setSize(WIDTH, HEIGHT);
+        scrollBar.setBorder(new HorizontalCustomBorder());
+
+        // paint image with thumb set to default value
+        BufferedImage image1 = new BufferedImage(WIDTH, HEIGHT, TYPE_INT_ARGB);
+        Graphics2D graphics2D = image1.createGraphics();
+        scrollBar.paint(graphics2D);
+        graphics2D.dispose();
+
+        // paint image with thumb set to max value
+        scrollBar.setValue(Integer.MAX_VALUE);
+        BufferedImage image2 = new BufferedImage(WIDTH, HEIGHT, TYPE_INT_ARGB);
+        Graphics2D graphics2D2 = image2.createGraphics();
+        scrollBar.paint(graphics2D2);
+        graphics2D2.dispose();
+
+        // check border for thumb
+        for (int i = WIDTH - BORDER_WIDTH; i < WIDTH; i++) {
+            for (int j = 0; j < HEIGHT; j++) {
```

Comment:
Now that there are two test methods, I'd suggest using `x` and `y` as loop variables:
```suggestion
        for (int x = WIDTH - BORDER_WIDTH; x < WIDTH; x++) {
            for (int y = 0; y < HEIGHT; y++) {
```

This would make it clearer how comparison works.